### PR TITLE
Update HVAC example to MySensors 2

### PR DESCRIPTION
**Pull request by Br3nda**
Other examples in the docs use MySensors 2, but this one was still for MySensor 1.x.

**Description:**
Update MySensors HVAC arduino sketch example to MySensors 2.x

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>
N/A

### CODE REVIEW

**Comment by MartinHjelmare** [comment]
Thanks! Instead of converting the sketch from 1.5 to 2.x, please keep the existing sketch and add another sketch with 2.x. That's how we've done it previously in other platforms when there's been an existing 1.5 sketch. There are still users who are on 1.5.

**Comment by Br3nda** [comment]
@MartinHjelmare Updated to include both 1.x and 2.x sketch

File: source/_components/climate.mysensors.markdown [hunk 1]
```diff
@@ -42,7 +42,138 @@ You can use V_TEMP to send the current temperature from the node to Home Assista
...
+        POWER_STATE = 1;
+        MODE_STATE = MODE_AUTO;
+      }
+      else if (recvData.equalsIgnoreCase("off")){
+        POWER_STATE = 0;
+      }
+      break;
+  }
+  sendHeatpumpCommand();
+}
```
Comment (MartinHjelmare) [inline source/_components/climate.mysensors.markdown]: You should feedback the new state to home assistant in the end of the receive method. Use the mysensors send method.

**Comment by Landrash** [comment]
@MartinHjelmare Will merge this when it got your approval. :+1:

**Comment by Landrash** [comment]
@Br3nda Any update on this?
@MartinHjelmare Mind fixing it yourself or should I close this?

**Comment by MartinHjelmare** [comment]
I could, but I can't really test this sketch.

**Comment by Br3nda** [comment]
I've added sending the change back to the gateway after actuating. Only lightly tested. There's a delay (as expected) but the UI seems to update eventually.

**Comment by MartinHjelmare** [comment]
@Br3nda the sketch looks good!

I'm interested in the delay. How long is that?

**Comment by Br3nda** [comment]
I'm now running in optimistic mode again because the delay is confusing in the UI.  Won't have time to debug and test for a bit. I did raise an issue over on the home-assistant repo a while back. If i get a weekend to test again I can comment there.

**Pull request merged**